src/main/helpers.js

```diff
@@ -17,6 +17,38 @@
 import stream from 'stream'
 import mime from 'mime-types'
 
+// Returns a wrapper function that will promisify a given callback function.
+// It will preserve 'this'.
+export function promisify(fn) {
+  return function() {
+    // If the last argument is a function, assume its the callback.
+    let callback = arguments[arguments.length - 1]
+    if (typeof callback !== 'function') callback = null
+
+    // If the callback is given, don't promisify, just pass straight in.
+    if (callback) return fn.apply(this, arguments)
```

Comment:
more concise:
if (typeof callback === 'function') return fn.apply(this, arguments)